cpp/src/io/parquet/error.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "error.hpp"
+
+#include <cudf/utilities/default_stream.hpp>
+
+#include <rmm/device_scalar.hpp>
+
+#include <sstream>
+
+namespace cudf::io::parquet {
+
+namespace {
+
+rmm::device_scalar<int32_t> error_code(0, cudf::get_default_stream());
```

Comment:
I don't think this should be global.

Suggestion:
class kernel_error{
rmm::device_scalar<int32_t> error_code;
public:
kernel_error(stream); // inits to zero
int32_t data();
int32_t value(); // use same stream as in ctor?
void reset(); // use same stream as in ctor?
operator std::string() const; // explicit?
};

Reply:
I originally went with global thinking we could avoid some memsets to 0. Initialize one at startup and then assume it stays zero since non zero will halt execution. But that didn't account for multiple invocations as I found while running the python tests. No need for it to be global now, since it has to be reset to zero before every use. Now it's just the time to allocate and free 4 bytes of device memory, but with pooled memory that shouldn't be a big deal.

Reply:
I'm scared of going global because of multi-threaded use cases. Opens the solution to many new challenges.